Fitbit.Portable/FitbitClient.cs

```diff
@@ -964,6 +908,164 @@ public Task<TimeSeriesDataListInt> GetTimeSeriesIntAsync(TimeSeriesResourceType
             }
         }
 
+        /// <summary>
+        /// Logs the specified Activity item for the current logged in user - https://dev.fitbit.com/docs/activity/#activity-logging
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>ActivityLog</returns>
+        public async Task<ActivityLog> LogActivityAsync(ActivityLog model)
+        {
+            string apiCall = FitbitClientHelperExtensions.ToFullUrl("/1/user/-/activities.json");
+
+            var items = new Dictionary<string, string>
+            {
+                {"activityName", Uri.EscapeDataString(model.Name)},
+                {"manualCalories", model.Calories.ToString()},
+                {"startTime", model.StartTime},
+                {"durationMillis", model.Duration.ToString()},
+                {"date", model.Date},
+                {"distance", model.Distance.ToString()}
+            };
+
+            if (!string.IsNullOrEmpty(model.ActivityId.ToString()) && model.ActivityId != 0)
+            {
+                items.Add("activityId", model.ActivityId.ToString());
+            }
+
+            apiCall = $"{apiCall}?{string.Join("&", items.Select(x => $"{x.Key}={x.Value}"))}";
+
+            HttpResponseMessage response = await HttpClient.PostAsync(apiCall, new StringContent(string.Empty));
+            await HandleResponse(response);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            return (new JsonDotNetSerializer() { RootProperty = "activityLog" }).Deserialize<ActivityLog>(responseBody);
+        }
+
+        /// <summary>
+        /// The Get Activity Logs List endpoint retrieves a list of a user's activity log entries before or after a given day with offset and limit using units in the unit system which corresponds to the Accept-Language header provided. - https://dev.fitbit.com/docs/activity/#get-activity-logs-list
+        /// </summary>
+        /// <param name="beforeDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to desc when using beforeDate.</param>
+        /// <param name="afterDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to asc when using afterDate.</param>
+        /// <param name="limit">The max of the number of entries returned (maximum: 20)</param>
+        /// <param name="encodedUserId">encoded user id, can be null for current logged in user</param>
+        /// <returns>ActivityLog</returns>
+        public async Task<List<ActivityLogsList>> GetActivityLogsList(DateTime? beforeDate, DateTime? afterDate, int limit = 20, string encodedUserId = default(string))
+        {
+            var apiCall = string.Empty;
+            limit = limit > 20 ? 20 : limit;

```

Comment:
Missing the check for negative values as per the original code review / pull request

Reply:
👍